File: src/calculate_Ergodicity.cpp

#include "calculate_Ergodicity.h"

using namespace Eigen;

double calculate_Ergodicity(MatrixXd ck, opt1 opt)
{
	double Ergodicity_Metric;
	
	Ergodicity_Metric=((opt.erg.LK).cwiseProduct(((ck-opt.erg.muk)).array().pow(2).matrix())).sum();

	return Ergodicity_Metric;
}

File: scripts/Calculate_Ergodicity.py
import numpy as np
from numpy import matrix


def Calculate_Ergodicity(Ergodicity_Metric,ck,opt):

	LK=opt.ergLK
	muk=opt.ergmuk

	Ergodicity_Metric=(matrix(np.multiply(LK,np.square(ck-muk)))).sum()

	return Ergodicity_Metric


